### DISCUSSION TITLE
CP-SAT engine version to version performance improvement report

### DISCUSSION THREAD

**Discussion by C8504** [google/or-tools#3290] [CP-SAT questions] [answered]
Hi experts,

Like commercial solvers, could you share with me general performance improvement about CP-SAT engine version to version on internal testsets or/and public test sets please?

As I tested on my testcases, I found v9.3 could find better solution than v9.2 with same timelimit.

Thanks.
John

**Comment by lperron** [comment, marked as answer]
Would be nice. But I do not trust the performance reports. 
They are usually over-tuning on a particular benchmark, and sometimes do not reflect in the user application. 

I prefer reporting on what we are working on.

**Reply by C8504** [reply]
Yes, I agree with you.
Could you share with us what you are working on reports and instances? Thanks

**Comment by lperron** [comment]
This is easy :-)

We are/have been working on:
  - scheduling (propagation, cuts, lower bounds)
  -  MaxSAT (presolve, core based heuristics)
  - MIP performance (presolve, cuts)

**Reply by C8504** [reply]
Prefect, hopefully you could share with us about v9.4 improvement on these parts.

**Reply by C8504** [reply]
BTW, could I know what "core based heuristics" is?

**Reply by lperron** [reply]
I will not go into deep details. Too long. You can look at the git commits. I try to write meaningful comments when it is interesting.

Core based search is a standard technique for SAT solvers.

**Reply by C8504** [reply]
Thanks lperron. Have a nice day.
